scripts/upload_demo_to_space.py

```diff
@@ -1,78 +1,159 @@
-                                colorFrom: indigo
-                                colorTo: indigo
-                                sdk: gradio
-                                sdk_version: {gradio_version}
-                                app_file: app.py
-                                pinned: false
-                                ---
-                                """
-            readme.open("w").write(textwrap.dedent(readme_content))
-
-        api = huggingface_hub.HfApi()
-        huggingface_hub.create_repo(
-            space_id,
+            ---
+            title: {space_id.split("/")[-1]} 
+            emoji: 💩
+            colorFrom: indigo
+            colorTo: indigo
+            sdk: gradio
+            sdk_version: {gradio_version}
+            app_file: app.py
+            pinned: false
+            ---
+            """
+            readme.write_text(textwrap.dedent(readme_content))
+
+        print("Uploading files to Hugging Face Space...")
+        api = HfApi()
+
+        print("Creating repository...")
+
+        # Create the repository if it doesn't exist
+        api.create_repo(
+            repo_id=space_id,
             space_sdk="gradio",
             repo_type="space",
             token=hf_token,
             exist_ok=True,
         )
-        api.upload_folder(
-            token=hf_token,
-            repo_id=space_id,
-            repo_type="space",
-            folder_path=tmpdir,
-            path_in_repo="",
-        )
+
+        print("Uploading files...")
+
+        # Collect all files in the temporary directory
+        all_files = sorted([p for p in pathlib.Path(tmpdir).rglob("*") if p.is_file()])
+
+        # Convert to relative paths for the repository
+        relative_files = [p.relative_to(tmpdir) for p in all_files]
+
+        # Create CommitOperationAdd objects for all files
+        operations = [
+            CommitOperationAdd(
+                path_in_repo=str(rel_path).replace(
+                    "\\", "/"
+                ),  # Ensure Unix-style paths
+                path_or_fileobj=str(pathlib.Path(tmpdir) / rel_path),
+            )
+            for rel_path in relative_files
+        ]
+
+        # Split operations into chunks of 50
+        operation_chunks = list(split_into_chunks(operations, 50))
+
+        for idx, chunk in enumerate(operation_chunks, start=1):
+            commit_message = f"Commit {idx}: Add {len(chunk)} file(s)"
+            try:
+                api.create_commit(
+                    repo_id=space_id,
+                    operations=chunk,
+                    commit_message=commit_message,
+                    token=hf_token,
+                    repo_type="space",
+                )
+                print(f"Successfully committed chunk {idx} with {len(chunk)} file(s).")
+            except Exception as e:
+                print(f"Failed to commit chunk {idx}: {e}")
+                raise e  # Re-raise the exception after logging
```

Comment:
```suggestion
                raise  # Re-raise the exception after logging
```

(nit) see https://stackoverflow.com/a/45986945